src/sorting/multithreaded_bogo_sort.rs

```diff
@@ -0,0 +1,92 @@
+use rand::prelude::*;
+use rand_chacha::ChaCha8Rng;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{mpsc, Arc};
+use std::thread;
+use std::time::Instant;
+
+const THREAD_COUNT: usize = 5;
+
+pub fn multithreaded_bogo_sort(data: &mut [i32]) {
+    // Start a counter to track performance
+    let clock = Instant::now();
+
+    #[warn(unused_variables)]
+    let (set_duration, _duration) = mpsc::channel();
+    let done = Arc::new(AtomicBool::new(false));
+
+    // Create a bunch of threads
+    let handles: Vec<_> = (0..THREAD_COUNT)
+        .map(|_| {
+            let set_duration = set_duration.clone();
+            let done = done.clone();
+            let mut local_data = data.to_owned();
+
+            thread::spawn(move || {
+                let mut rng = ChaCha8Rng::seed_from_u64(thread_rng().gen());
+
+                while !done.load(Ordering::Acquire) {
+                    local_data.shuffle(&mut rng);
+
+                    if local_data.windows(2).all(|w| w[0] <= w[1]) {
+                        set_duration.send(clock.elapsed()).unwrap();
+                        done.store(true, Ordering::Release);
+                        break;
+                    }
+                }
+            })
+        })
+        .collect();
+
+    // To know how long it took to sort, uncomment the following lines:
+    // let elapsed_time = duration.recv().unwrap();
+    // println!("{:?}", elapsed_time);
+
+    // Join the handles in the vector
+    for handle in handles {
+        handle.join().unwrap();
+    }
+
+    // Sort the original data in-place using the last sorted result
+    data.sort();
```

Comment:
What do you mean by `last sorted result` here? How do sorting threads mutate `data` array?

Reply:
The function creates multiple threads, each with a local copy of the data array. Threads repeatedly shuffle their local copies until the entire array is sorted or a global condition is met.

Reply:
If they are all working with a local copy, how does the original array get sorted? We can just call `data.sort()` and don't bother with arrays if they don't change `data`

Reply:
Right. I'll modify it.

Reply:
Until then, can you please look into the simpson method?